internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/BlobP2PTransferAmongServersTest.java

```diff
@@ -0,0 +1,182 @@
+import static com.linkedin.davinci.store.AbstractStorageEngine.METADATA_PARTITION_ID;
+import static com.linkedin.davinci.store.rocksdb.RocksDBServerConfig.ROCKSDB_PLAIN_TABLE_FORMAT_ENABLED;
+import static com.linkedin.venice.integration.utils.VeniceClusterWrapper.DEFAULT_KEY_SCHEMA;
+import static com.linkedin.venice.utils.IntegrationTestPushUtils.defaultVPJProps;
+import static com.linkedin.venice.vpj.VenicePushJobConstants.VENICE_STORE_NAME_PROP;
+
+import com.linkedin.venice.ConfigKeys;
+import com.linkedin.venice.controllerapi.ControllerClient;
+import com.linkedin.venice.controllerapi.UpdateStoreQueryParams;
+import com.linkedin.venice.integration.utils.ServiceFactory;
+import com.linkedin.venice.integration.utils.VeniceClusterWrapper;
+import com.linkedin.venice.integration.utils.VeniceServerWrapper;
+import com.linkedin.venice.meta.PersistenceType;
+import com.linkedin.venice.offsets.OffsetRecord;
+import com.linkedin.venice.store.rocksdb.RocksDBUtils;
+import com.linkedin.venice.utils.IntegrationTestPushUtils;
+import com.linkedin.venice.utils.TestUtils;
+import com.linkedin.venice.utils.TestWriteUtils;
+import com.linkedin.venice.utils.Utils;
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+import org.apache.commons.io.FileUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.testng.Assert;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+
+public class BlobP2PTransferAmongServersTest {
+  private static final Logger LOGGER = LogManager.getLogger(BlobP2PTransferAmongServersTest.class);
+
+  private String path1;
+  private String path2;
+  private VeniceClusterWrapper cluster;
+
+  @BeforeMethod
+  public void setUp() {
+    cluster = initializeVeniceCluster();
+  }
+
+  @AfterMethod
+  public void tearDown() {
+    Utils.closeQuietlyWithErrorLogged(cluster);
+  }
+
+  @Test(singleThreaded = true)
+  public void testBlobP2PTransferAmongServers() throws Exception {
+    String storeName = "test-store";
+    Consumer<UpdateStoreQueryParams> paramsConsumer = params -> params.setBlobTransferEnabled(true);
+    setUpStore(cluster, storeName, paramsConsumer, properties -> {}, true);
+
+    VeniceServerWrapper server1 = cluster.getVeniceServers().get(0);
+    VeniceServerWrapper server2 = cluster.getVeniceServers().get(1);
+
+    // very the snapshot is generated for both servers after the job is done
+    for (int i = 0; i < 3; i++) {
+      String snapshotPath1 = RocksDBUtils.composeSnapshotDir(path1 + "/rocksdb", storeName + "_v1", i);
+      Assert.assertTrue(Files.exists(Paths.get(snapshotPath1)));
+      String snapshotPath2 = RocksDBUtils.composeSnapshotDir(path2 + "/rocksdb", storeName + "_v1", i);
+      Assert.assertTrue(Files.exists(Paths.get(snapshotPath2)));
+    }
```

Comment:
A dumb question, do we remove snapshot path after the blob transfer is done in the design?

Reply:
I missed this. Yes, @jingy-li please double-check and see if we need to do more clean-up here

Reply:
In server side, we do not clean the snapshot when the transfer is done.